src/schoology-extractor/schoology_extractor/schoology_extract_facade.py

```diff
@@ -113,12 +113,11 @@ def get_assignments(self, section_id: int) -> pd.DataFrame:
         pd.DataFrame
             DataFrame with all assignment data, in the unified data model format.
         """
+        return pd.DataFrame(self._client.get_assignments(section_id, self._page_size))
 
-        assignments = self._client.get_assignments(section_id, self._page_size)
-        df = pd.DataFrame(assignments)
-        df["section_id"] = section_id
-
-        return assignmentsMap.map_to_udm(df)
+    def map_assignments_to_udm(self, assignments: pd.DataFrame, section_id: Union[int, str]):
+        assignments["section_id"] = section_id
+        return assignmentsMap.map_to_udm(assignments)

```

Comment:
I don't see the value in splitting this up.

Reply:
get_submissions method from facade, receives the assignments in their original shape, so I hadto get the original data, assign it to the result_bucket in main, and then map it to udm.

```python3
assignments = service.get_assignments(section_id)
result_bucket["assignments"] = assignments
return service.map_assignments_to_udm(assignments, section_id) if not assignments.empty else None
```

Reply:
And I could not use the udm version, because get_submissions needs access to grade_item_id, which  is not part of UDM version, it is part of the original API response.

Reply:
OH, I see now.